includes/class-wc-reepay-admin.php

```diff
@@ -0,0 +1,452 @@
+				),
+				'',
+				dirname( __FILE__ ) . '/../templates/'
+			);
+		}
+	}
+
+	/**
+	 * Inserts the content of the API actions into the meta box
+	 */
+	public function meta_box_payment() {
+		global $post;
+		global $post_id;
+
+		$order = wc_get_order( $post->ID );
+
+		if ( $order ) {
+			$payment_method = $order->get_payment_method();
+			if ( in_array( $payment_method, WC_ReepayCheckout::PAYMENT_METHODS ) ) {
+
+				do_action( 'woocommerce_reepay_meta_box_payment_before_content', $order );
+
+				$order = wc_get_order( $post_id );
+				$gateway = rp_get_payment_method( $order );
+				$order_data = $gateway->api->get_invoice_data( $order );
+				if ( is_wp_error( $order_data ) ) {
+					return;
+				}
+
+				wc_get_template(
+					'admin/metabox-order.php',
+					array(
+						'gateway'    => $gateway,
+						'order'      => $order,
+						'order_id'   => $order->get_id(),
+						'order_data' => $order_data
+					),
+					'',
+					dirname(__FILE__) . '/../templates/'
+				);
+			}
+		}
+	}
+
+	public function meta_box_subscription() {
+		$this->meta_box_payment();
+	}
+
+	/**
+	 * @param WC_Order $order
+	 */
+	public static function add_action_buttons( $order ) {
+		$payment_method = $order->get_payment_method();
+		if ( in_array( $payment_method, WC_ReepayCheckout::PAYMENT_METHODS ) ) {
+			$gateway = rp_get_payment_method( $order );
+
+			wc_get_template(
+				'admin/action-buttons.php',
+				array(
+					'gateway'    => $gateway,
+					'order'      => $order
+				),
+				'',
+				dirname( __FILE__ ) . '/../templates/'
+			);
+		}
+	}
+
+	/**
+	 * Enqueue Scripts in admin
+	 *
+	 * @param $hook
+	 *
+	 * @return void
+	 */
+	public static function admin_enqueue_scripts( $hook ) {
+		if ( $hook === 'post.php' ) {
+			// Scripts
+			$suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
+
+			wp_register_script(
+				'reepay-js-input-mask',
+				plugin_dir_url( __FILE__ ) . '../assets/js/jquery.inputmask' . $suffix . '.js',
+				array( 'jquery'),
+				'5.0.3'
+			);
+
+			wp_register_script(
+				'reepay-admin-js',
+				plugin_dir_url( __FILE__ ) . '../assets/js/admin' . $suffix . '.js',
+				array(
+					'jquery',
+					'reepay-js-input-mask'
+				)
+			);
+
+			wp_enqueue_style( 'wc-gateway-reepay-checkout', plugins_url( '/../assets/css/style' . $suffix . '.css', __FILE__ ), array(), false, 'all' );
+
+			// Localize the script
+			$translation_array = array(
+				'ajax_url'  => admin_url( 'admin-ajax.php' ),
+				'text_wait' => __( 'Please wait...', 'woocommerce-gateway-reepay-checkout' ),
+				'nonce'     => wp_create_nonce( 'reepay' ),
+			);
+			wp_localize_script( 'reepay-admin-js', 'Reepay_Admin', $translation_array );
+
+			// Enqueued script with localized data
+			wp_enqueue_script( 'reepay-admin-js' );
+		}
+	}
+
+	/**
+	 * Action for Capture
+	 */
+	public function ajax_reepay_capture() {
+		if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'reepay' ) ) {
+			exit( 'No naughty business' );
```

Comment:
When closing the popup and canceling the order, users are redirected to an empty page with a strange inscription “No naughty business” https://prnt.sc/VFF-3A2kMi3A, this text is also displayed in the code in 5 more places https://prnt.sc/KGHSR8HkxROZ, they are displayed when the nonce check fails, but during the test I was transferred to such a page

Reply:
Fixed